File: Data Structures & Algorithms/max-water-container/submission-1.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) {
        var l = 0
        var r = heights.length - 1

        var res = 0

        while (l < r){
            res = Math.max(res, Math.min(heights[l], heights[r]) * (r - l))

            if (heights[l] < heights[r]){
                l ++;
            } else{
                r --;
            }
        }

        return res
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l = 0
        r = len(nums) - 1
        res = math.inf

        while l <= r:
            m = (r + l) // 2

            if nums[m] >= nums[l] >= nums[r]:
                l = m + 1
            elif nums[l] <= nums[m] <= nums[r]:
                r = m - 1
            elif nums[m] <= nums[r] <= nums[l]:
                l += 1
            
            res = min(res, nums[m])

        return res
            
            
        